src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    // const join2 = [...this];
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === 0) {
+        if (this[i] === undefined || this[i] === null) {
+          result += '';
+        } else {
+          result += this[i];
+        }
+      }
+
+      if (i > 0) {
+        if (this[i] === undefined || this[i] === null) {
+          result = result + separator + '';
+        } else {
+          result = result + separator + this[i];
+        }
+      }
```

Comment:
try to avoid nested if and if/else statements.
you need just two if

- current item not equal undefined and not equal null - in this case add current item to result
- current item not last item - in this case ass separator to result.